source/extensions/omni.isaac.lab/omni/isaac/lab/scene/interactive_scene.py

```diff
@@ -333,6 +333,38 @@ def extras(self) -> dict[str, XFormPrimView]:
         """
         return self._extras
 
+    @property
+    def state(self) -> dict[str, dict[str, torch.Tensor]]:
+        """Returns the state of the scene entities.
+
+        Returns:
+            A dictionary of the state of the scene entities.
+        """
+        return self.get_state(is_relative=False)
+
+    def get_state(self, is_relative: bool = False) -> dict[str, dict[str, torch.Tensor]]:
+        """Returns the state of the scene entities.
+
+        Returns:
+            A dictionary of the state of the scene entities.
+        """
+        state = dict()
+        # -- assets
+        state["articulation"] = dict()
+        for asset_name, articulation in self._articulations.items():
```

Comment:
it would be good to get velocity and applied_torque too for more general use cases.

Reply:
Added velocity, together with position, to the articulation get/reset_to state.
To make the obtained state (to be recorded in a dataset) future-proof, I also updated to store an asset's state in a dict, in contrast to storing an asset's state in a tensor that needs additional interpretation, so we can add new key-tensor entries to the state in later revision while maintaining recorded dataset's backward-compatibility. 
That is, instead of storing a robot's (or a rigid object's) state in one tensor, now the state is stored in a dict like this:
```
state["articulation"][asset_name]["joint_position"] = articulation.data.joint_pos.clone()
state["articulation"][asset_name]["joint_velocity"] = articulation.data.joint_vel.clone()
```

Reply:
do we need to capture the root position and velocity as well?

Reply:
Added root state.